### PR TITLE
sdr-ui: FSPL distance estimator in the Radio panel (closes #164)

### LINKED ISSUE #164: Distance estimation from signal strength (FSPL calculator)
Estimate transmitter distance from received signal strength using the Free-Space Path Loss (FSPL) formula:

```
FSPL(dB) = 20*log10(d) + 20*log10(f) - 147.55
```

Where d = distance (meters), f = frequency (Hz). Rearranging for distance:

```
d = 10^((FSPL - 20*log10(f) + 147.55) / 20)
```

Requirements:
- User inputs transmitter ERP (effective radiated power) — public safety radios are typically 25-50W
- Calibrated receiver reference level (dBm) — would need a known-power reference signal or manual calibration offset
- Frequency is already known from the tuner
- Signal level is already measured in the signal history graph (#150)

Could integrate with RadioReference (#151) to auto-fill ERP for known transmitters. Display estimated distance in the status bar or as an overlay on the signal history graph.

Note: FSPL is idealized (line-of-sight, no multipath/terrain). Real-world accuracy depends heavily on environment — urban vs rural, terrain, antenna height, obstructions. Best used as a rough estimate, not precision ranging.

### PR DESCRIPTION
**Pull request by jasonherald**
## Summary

Adds a "Distance Estimator" section at the bottom of the Radio sidebar panel. Given a transmitter ERP and a receiver calibration offset, it runs the textbook free-space path loss formula against the live signal level and tuner frequency to show an estimated distance.

\`\`\`
FSPL(dB) = 20·log10(d) + 20·log10(f) - 147.55
        ⇒ d = 10 ^ ((FSPL - 20·log10(f) + 147.55) / 20)
\`\`\`

The \`147.55\` dB constant is \`20·log10(c / 4π)\` — computed analytically from the exact-by-definition speed of light so the formula matches any textbook derivation.

### \`sdr-dsp::propagation\` (new module)

Pure-math helpers, no threading or I/O:

- \`fspl_distance_m(erp_dbm, received_dbm, frequency_hz) -> f64\`
- \`fspl_db(distance_m, frequency_hz) -> f64\`  (round-trip helper)
- \`watts_to_dbm\` / \`dbm_to_watts\`

**8 unit tests** — including a round-trip across 5 frequencies × 5 distances, textbook anchors (1 W = 30 dBm, FSPL at 100 MHz / 1 m = 12.45 dB, etc.), NaN/zero/negative edge cases, and a "50 W @ 155 MHz @ -90 dBm ≈ 1000 km" sanity anchor.

### Radio panel UI

New \`AdwPreferencesGroup\` "Distance Estimator" with three rows:
- **Transmitter Power** — \`AdwSpinRow\`, watts, default 25 W (mobile public-safety). Range 0.001–100 000 W.
- **Receiver Calibration** — \`AdwSpinRow\`, dB, default 0. Range ±30 dB for any reasonable RTL-SDR scale offset.
- **Estimated Distance** — read-only \`AdwActionRow\`, subtitle updates live.

Display state is explicit as an enum with five variants:

| State | Trigger | Display |
|---|---|---|
| \`NoData\` | no signal level has ever flowed | \`—\` |
| \`NoActiveSignal\` | calibrated received < -130 dBm | \`No active signal\` |
| \`CheckCalibration\` | received ≥ transmitted (physically impossible) | \`Check calibration\` |
| \`TooWeak\` | FSPL > Earth's great-circle max (~20 000 km) | \`Too weak to measure\` |
| \`Value(d)\` | meaningful | \`X m\` / \`X.X km\` / \`XX km\` |

The -130 dBm threshold sits just below a sensitive narrowband receiver's MDS — below it the signal is almost certainly squelch-gated or noise-floor and the FSPL math would saturate at physically absurd distances. **12 unit tests** cover every state transition plus the number formatting.

### Wiring (window.rs)

- \`DspToUi::SignalLevel\` handler calls \`radio_panel.update_distance_from_signal(level, center_freq)\` on every level update (~10 Hz)
- \`freq_selector.connect_frequency_changed\` calls \`update_distance_frequency\` so the estimate tracks retunes
- ERP / calibration spin rows have two value-notify handlers: one inside \`build_radio_panel\` recomputes the display from cached inputs; one in \`connect_distance_estimator_persistence\` writes to \`ConfigManager\`
- Initial values seeded from config at startup, falling back to 25 W / 0 dB on a fresh config

Config keys (public constants on \`radio_panel\`):
- \`radio_distance_erp_watts\`
- \`radio_distance_calibration_db\`

## Caveat (documented in-panel)

The group subtitle reads:

> *Rough line-of-sight (FSPL) estimate — read as an upper bound, not precision ranging*

FSPL is the idealised line-of-sight loss between two isotropic antennas in free space — no terrain, no buildings, no multipath, no atmospheric effects. Real-world receive levels drop off faster, so the estimate is really "the transmitter is *at most* this far away for this received level." Helpful for antenna adjustment / propagation sanity-check, not for precision ranging.

## Scope explicitly out

- **RadioReference ERP auto-fill** (ticket suggests this → #151 area). Not in this PR.
- **Overlay on the signal-history graph.** Status-row display in the Radio panel is sufficient; graph overlay can be a follow-up if you want it.
- **Terrain / multipath modelling.** FSPL is deliberately the simplest estimate.

## User smoke test

Verified on 25 W public-safety NFM at 155 MHz:
- Real signal shows ~9 km (matches hand-computed FSPL)
- Squelch-gated / no-signal state shows **"No active signal"** (replaces the earlier "out of range (signal below noise floor?)" message per user feedback)
- ERP / calibration knobs update the distance display immediately on edit
- Persistence across restart works (25 W / 0 dB defaults on fresh config)
- Rest of Radio panel (bandwidth / squelch / filters / de-emphasis / CTCSS) regression-free; demod visibility toggling unaffected

## Test plan

- [x] \`cargo test -p sdr-dsp --lib propagation\` — 8/8 pass + doctest
- [x] \`cargo test -p sdr-ui --lib --features whisper-cpu distance_display\` — 12/12 pass
- [x] \`cargo clippy --all-targets --workspace --features whisper-cpu -- -D warnings\` clean
- [x] \`cargo fmt --all -- --check\` clean
- [x] \`make install CARGO_FLAGS=\"--release --features whisper-cuda\"\` — user smoke-tested on live signals

Closes #164.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

* **New Features**
  * Added an Estimated Distance estimator in the radio sidebar with ERP (transmitter power) and receiver calibration controls; estimates auto-update from received signal level and tuner frequency and persist between sessions.
  * Exposed public propagation math helpers for FSPL, power/unit conversions, and distance estimation for use by other components.

* **Tests**
  * Added unit tests covering numeric anchors, edge cases, and UI transition/formatting logic.

### CODE REVIEW

File: crates/sdr-ui/src/sidebar/radio_panel.rs [hunk 1]
```diff
@@ -486,6 +564,306 @@ impl RadioPanel {
...
+            Self::TooWeak => "Too weak to measure".to_string(),
+            Self::Value(d) => {
+                if d < 1_000.0 {
+                    format!("{} m", d.round() as u64)
+                } else if d < 1_000_000.0 {
+                    format!("{:.1} km", d / 1_000.0)
+                } else {
+                    let km_rounded_10 = ((d / 1_000.0) / 10.0).round() * 10.0;
+                    format!("{km_rounded_10:.0} km")
+                }
```
[the remark this answers is not in the document]
Comment (jasonherald) [reply crates/sdr-ui/src/sidebar/radio_panel.rs:752]: Done in `fc3b6ac` — extracted four named constants matching the antenna.rs pattern:

- `KM_THRESHOLD_M = 1_000.0` (m → km boundary)
- `MEGAMETRE_THRESHOLD_M = 1_000_000.0` (switch to km-rounded-to-10)
- `MEGAMETRE_ROUND_KM = 10.0` (the rounding granularity — doc explains why)
- `METRES_PER_KM = 1_000.0` (avoid magic-literal reuse with the threshold)

Each has a one-line rationale docstring. The match-arm is now literal-free.

File: crates/sdr-dsp/src/propagation.rs [hunk 2]
```diff
@@ -0,0 +1,269 @@
...
+/// formulas match any textbook derivation precisely.
+///
+/// `const fn`-free because `log10` isn't a const-fn; we compute it
+/// lazily via `Lazy` in a helper accessor instead of forcing the
+/// computation inline every call.
+#[inline]
+fn fspl_constant_db() -> f64 {
+    // 20 · log10(c / (4π))
+    20.0 * (C_M_PER_S / (4.0 * PI)).log10()
+}
```
[the remark this answers is not in the document]
Comment (jasonherald) [reply crates/sdr-dsp/src/propagation.rs]: Fixed in `fc3b6ac`. Rewrote the doc to describe what the code actually does — inline recomputation, cheap at our <=60 Hz call rate, not worth a `OnceCell` dep. Kept the implementation as-is (no behavior change).

File: crates/sdr-ui/src/sidebar/radio_panel.rs [hunk 3]
```diff
@@ -486,6 +564,327 @@ impl RadioPanel {
...
+
+    #[test]
+    fn format_exactly_at_cap_still_shown_as_number() {
+        // Boundary case: the cap value itself is still a real
+        // distance (half of Earth's circumference). Only values
+        // STRICTLY above it transition to `TooWeak`.
+        let formatted = DistanceDisplay::Value(MAX_MEANINGFUL_DISTANCE_M).format();
+        assert!(formatted.ends_with(" km"));
+    }
 }
```
[the remark this answers is not in the document]
Comment (jasonherald) [reply crates/sdr-ui/src/sidebar/radio_panel.rs]: Done in `3d9a38b` — all 12 distance state-machine + format tests now live under the existing bottom `mod tests` block, alongside the compile-time invariant const-asserts for every panel constant group (bandwidth / squelch / NB / notch / ERP / calibration). No separate test module mid-file. 199/199 tests pass.

Also piggy-backed a user-requested optimization in the same commit: visibility-gate the distance calc so `update_distance_from_signal` / `update_distance_frequency` skip the FSPL compute + `set_subtitle` call when the Radio panel isn't mapped (user on a different activity tab). Inputs are still cached so the display re-syncs on the next `connect_map` when they switch back.

File: crates/sdr-ui/src/sidebar/radio_panel.rs [hunk 4]
```diff
@@ -94,6 +98,59 @@ const CTCSS_THRESHOLD_STEP: f64 = 0.01;
...
+const DEFAULT_CALIBRATION_DB: f64 = 0.0;
+/// Minimum calibration offset. ±30 dB covers every reasonable
+/// RTL-SDR reference-level scenario we've seen.
+const MIN_CALIBRATION_DB: f64 = -30.0;
+/// Maximum calibration offset.
+const MAX_CALIBRATION_DB: f64 = 30.0;
+/// Calibration offset step.
+const CALIBRATION_STEP_DB: f64 = 0.5;
+/// Calibration offset page step.
+const CALIBRATION_PAGE_DB: f64 = 5.0;
```
[the remark this answers is not in the document]
Comment (jasonherald) [reply crates/sdr-ui/src/sidebar/radio_panel.rs:152]: Done in `3d9a38b` — added two const-assert blocks in the bottom `mod tests` matching the bandwidth/squelch/NB/notch pattern:

```rust
const _: () = {
    assert!(super::MIN_ERP_WATTS <= super::MAX_ERP_WATTS);
    assert!(super::DEFAULT_ERP_WATTS >= super::MIN_ERP_WATTS);
    assert!(super::DEFAULT_ERP_WATTS <= super::MAX_ERP_WATTS);
    assert!(super::ERP_STEP_WATTS > 0.0);
    assert!(super::ERP_PAGE_WATTS > 0.0);
};
```

Same shape for calibration.
